Realm/ObjectServerTests/SwiftCollectionSyncTests.swift

```diff
@@ -338,3 +338,602 @@ class CollectionSyncTestCase: SwiftSyncTestCase {
+            } else if case .float = value {
+                XCTAssertEqual(o.anyCol, .double(Double(value.floatValue!)))
+            } else if case .date = value {
+                // Date is not exact when synced
+                let date = o.anyCol.dateValue
+                XCTAssertTrue(Calendar.current.isDate(date!, equalTo: value.dateValue!, toGranularity: .second))
+            } else {
+                XCTAssertEqual(o.anyCol, value)
+            }
+        }
+    }
+
+    @MainActor func testSyncMixedArray() async throws {
+        let list = sampleData
+        try await write { realm in
+            let object = SwiftTypesSyncObject()
+            object.anyCol = AnyRealmValue.fromArray(list)
+            object.stringCol = self.name
+            realm.add(object)
+        }
+
+        let realm = try await openRealm()
+        let results0 =  try await realm.objects(SwiftTypesSyncObject.self).where { $0.stringCol == name }.subscribe()
+        XCTAssertEqual(results0.count, 1)
+
+        let o = results0.first!
+        for (index, value) in list.enumerated() {
+            assertListEqual(o, index, value)
+        }
+    }
+
+    @MainActor func testSyncMixedDictionary() async throws {
+        let dictionary = Dictionary(uniqueKeysWithValues: sampleData.enumerated().map { ("\($0)", $1) })
+        try await write { realm in
+            let object = SwiftTypesSyncObject()
+            object.anyCol = AnyRealmValue.fromDictionary(dictionary)
+            object.stringCol = self.name
+            realm.add(object)
+        }
+
+        let realm = try await openRealm()
+        let results0 =  try await realm.objects(SwiftTypesSyncObject.self).where { $0.stringCol == name }.subscribe()
+        XCTAssertEqual(results0.count, 1)
+
+        let o = results0.first!
+
+        for (key, value) in dictionary {
+            assertDictionaryEqual(o, key, value)
+        }
+    }
+
+    @MainActor func testSyncMixedNestedArray() async throws {
+        let so = SwiftPerson()
+        so.firstName = "Doe"
+        let subArray3: AnyRealmValue = AnyRealmValue.fromArray([ .object(so), .double(123.456) ])
+        let subArray2: AnyRealmValue = AnyRealmValue.fromArray([ subArray3 ])
+        let subArray1: AnyRealmValue = AnyRealmValue.fromArray([ subArray2 ])
+        let array: Array<AnyRealmValue> = [
+            subArray1, .bool(false)
+        ]
+        try await write { realm in
+            let object = SwiftTypesSyncObject()
+            object.anyCol = AnyRealmValue.fromArray(array)
+            object.stringCol = self.name
+            realm.add(object)
+        }
+
+        let realm = try await openRealm()
+        let results0 =  try await realm.objects(SwiftTypesSyncObject.self).where { $0.stringCol == name }.subscribe()
+        XCTAssertEqual(results0.count, 1)
+
+        let o = results0.first!
+        XCTAssertEqual(o.anyCol.listValue?[1], .bool(false))
```

Comment:
Nitpick: Defining a local variable for `o.anyCol.listValue?` would make it a bit more readable.